src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketDirectory.java

```diff
@@ -0,0 +1,51 @@
+package com.atlassian.bitbucket.jenkins.internal.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @since 3.0.0
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BitbucketDirectory {
+
+    private BitbucketPage<BitbucketDirectoryChild> children;
+    private BitbucketDirectoryPath path;
+    private String revision;
+
+    @JsonCreator
+    public BitbucketDirectory(
+            @JsonProperty(value = "children") BitbucketPage<BitbucketDirectoryChild> children,
+            @JsonProperty(value = "path") BitbucketDirectoryPath path,
+            @JsonProperty(value = "revision") String revision) {
+        this.children = children;
+        this.path = path;
+        this.revision = revision;
+    }
+
+    public BitbucketPage<BitbucketDirectoryChild> getChildren() {
+        return children;
+    }
+
+    public void setChildren(
```

Comment:
There's a couple of what seems to be unused methods in `BitbucketDirectory`, `BitbucketDirectoryChild`, `BitbucketDirectoryPath`. Are the setters required for JSON serialisation? Or is it only the getters?

Reply:
I think these are unnecessary. I'll remove, thanks for catching :)